packages/validation/src/rule-provider.ts

```diff
@@ -234,7 +236,9 @@ export class PropertyRule<TObject extends IValidateable = IValidateable, TValue
   public withMessage(message: string) {
     const rule = this.latestRule;
     this.assertLatestRule(rule);
-    this.messageProvider.setMessage(rule, message);
+    // eslint-disable-next-line no-console
+    if (__DEV__ && rule._isStateful) { console.debug('Setting message to stateful rule will override the message mapper set to the rule.'); }
```

Comment:
can we do a direct check instead? Trying to think about this in an abstract way with a private property will cause confusion.
```suggestion
    if (__DEV__ && rule instanceof StateRule) { console.debug('Setting message to stateful rule will override the message mapper set to the rule.'); }
```